app/src/app/layout/table/table.component.ts

```diff
@@ -103,8 +103,11 @@ export class TableComponent<T> implements OnInit, OnDestroy {
   @Input('columns')
   public set Columns(columns: Array<Column>) {
     this.columns = columns;
-    this.initialColumns = Object.assign([], columns);
-    this.items = [];
+    const temp = [];
```

Comment:
In my opinion this is not the right solution. Because Object Assign only makes a shallow copy, copying the properties one by one is a nice hack, but in my opinion it does not solve the problem of references, because if you continue to nest the elements (e.g. the render function) this concept does not work anymore. I think we should rather store only a list of visible columns as reset value. Similar to the states.

Reply:
Actually, we don't want to reset the whole table, but only the displayed columns. Therefore it is actually enough if we save a list with variables of the visible columns and then adjust them to the hidden columns. 
Something like:
```
for (const col of this.columns) {
        col.hidden = !visibleColumns.includes(col.name);
}
```